torchao/_models/llama/tp.py

```diff
@@ -0,0 +1,162 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+
+# copied from https://github.com/pytorch-labs/gpt-fast/blob/main/tp.py
+import os
+from typing import List, Optional
+
+import torch
+import torch.distributed as dist
+from torch import nn
+if os.uname().sysname != "Darwin":
+    from torch.distributed import _functional_collectives as funcol
+else:
+    # Distributed is not supported on MacOS
+    funcol = None
+
+from torchao._models.llama.model import Attention, FeedForward, Transformer
+from torchao.quantization.GPTQ import WeightOnlyInt4Linear
```

Comment:
we are not using`WeightOnlyInt4Linear` any more in torchao I think, is this just for GPTQ?